[Profiler] Add TSC Clock Callback to CUPTI

sraikund16 opened the pull request:
Summary:
Right now we use the default clock for CUPTI which is not monotonic nor particularly fast. We have already added the Kineto side of the implementation here: https://www.internalfb.com/diff/D56525885

This diff only adds the compile flags such that the TSC format is used and sets the converter using a libkineto call in the profiler

Test Plan:
Obtained following trace using resnet test:
https://www.internalfb.com/intern/perfdoctor/trace_view?filepath=tree/traces/dynocli/devvm2185.cco0.facebook.com/rank-0.Apr_25_11_03_18.3862943.pt.trace.json.gz&bucket=gpu_traces

TBD: Add benchmarks

Differential Revision: D56584521

aaronenyeshi reviewed (changes requested):
hold on this feature, until we have better CI set-up

sraikund16 committed:
[Profiler] Add TSC Clock Callback to CUPTI (#125036)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/125036

Right now we use the default clock for CUPTI which is not monotonic nor particularly fast. We have already added the Kineto side of the implementation here: https://www.internalfb.com/diff/D56525885

This diff only adds the compile flags such that the TSC format is used and sets the converter using a libkineto call in the profiler

Test Plan:
Obtain results as shown in following doc:
https://docs.google.com/document/d/1d-OSi-qdmLFmN8lDMPFr77uT8BgAhwgwIq_eM21HNnY/edit#heading=h.eu8d3rcongce

Differential Revision: D56584521
